contracts/FeralfileArtworkV3.sol

```diff
@@ -0,0 +1,433 @@
+        internal
+    {
+        require(
+            _exists(transferParam_.tokenID),
+            "ERC721: artwork edition is not found"
+        );
+
+        require(
+            _isApprovedOrOwner(transferParam_.from, transferParam_.tokenID),
+            "ERC721: caller is not token owner nor approved"
+        );
+
+        require(
+            transferParam_.timestamp + 5 minutes >= block.timestamp,
+            "FeralfileExhibitionV3: timestamp is over recv window"
+        );
+
+        bytes32 requestHash = keccak256(
+            abi.encode(
+                transferParam_.from,
+                transferParam_.to,
+                transferParam_.tokenID,
+                transferParam_.timestamp
+            )
+        );
+
+        require(
+            isValidRequest(
+                requestHash,
+                transferParam_.r_,
+                transferParam_.s_,
+                transferParam_.v_,
+                transferParam_.tokenID
+            ),
+            "FeralfileExhibitionV3: the transfer request is not authorized"
+        );
+
+        _safeTransfer(
+            transferParam_.from,
+            transferParam_.to,
+            transferParam_.tokenID,
+            ""
+        );
+    }
+
+    /// @notice batchMint is function mint array of tokens
+    /// @param mintParams_ - the array of transfer parameters
+    function batchMint(MintArtworkParam[] memory mintParams_)
+        external
+        onlyAuthorized
+    {
+        for (uint256 i = 0; i < mintParams_.length; i++) {
+            _mintArtwork(
+                mintParams_[i].artworkID,
+                mintParams_[i].edition,
+                mintParams_[i].artist,
+                mintParams_[i].owner,
+                mintParams_[i].ipfsCID
+            );
+        }
+    }
+
+    /// @notice mint artwork to ERC721
+    /// @param artworkID_ - the artwork id where the new edition is referenced to
+    /// @param editionNumber_ - the edition number of the artwork edition
+    /// @param artist_ - the artist address of the new minted token
+    /// @param owner_ - the owner address of the new minted token
+    /// @param ipfsCID_ - the IPFS cid for the new token
+    function _mintArtwork(
+        uint256 artworkID_,
+        uint256 editionNumber_,
+        address artist_,
+        address owner_,
+        string memory ipfsCID_
+    ) internal {
+        /// @notice the edition size is not set implies the artwork is not created
+        require(
+            artworks[artworkID_].editionSize > 0,
+            "FeralfileExhibitionV3: artwork is not found"
+        );
+        /// @notice The range of editionNumber should be between 0 (AP) ~ artwork.editionSize
+        require(
+            editionNumber_ <= artworks[artworkID_].editionSize,
+            "FeralfileExhibitionV3: edition number exceed the edition size of the artwork"
+        );
+        require(owner_ != address(0), "invalid owner address");
```

Comment:
```suggestion
        require(owner_ != address(0), "invalid owner address");
        require(artist_ != address(0), "invalid artist address");
```
If we do not accept artist in zero address.

Reply:
Added